### PR TITLE
feat: add test tags

### PR DESCRIPTION
**Pull request by kafkaphoenix**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [ ] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by dfinteligenz** [comment]
For making this change work, you'd need to change the tests github workflow so that it executes both unit and integration tests in separate jobs like below:

```
- name: Run unit tests
        if: matrix.changes == 'true'
        run: go test ./... -cover -v -coverpkg=./... -coverprofile=coverage-unit.out --tags=unit
        working-directory: ./engine/${{ matrix.component }}
      - name: Run integration tests
        if: matrix.changes == 'true'
        run: go test ./... -cover -v -coverpkg=./... -coverprofile=coverage-integration.out --tags=integration
        working-directory: ./engine/${{ matrix.component }}
```

Checkout the kai github workflow to see a complete example: https://github.com/konstellation-io/kai-kli/blob/997b8512819274f3b68bf9760c1d97f6b96759bf/.github/workflows/test-release.yml#L44

File: .github/workflows/test-release.yml [hunk 1]
```diff
@@ -41,16 +41,27 @@ jobs:
...
         run: |
-          go test -coverprofile="${{ matrix.platform }}_coverage.out" -covermode=atomic -coverpkg=./... ./...
-          go tool cover -func="${{ matrix.platform }}_coverage.out"
+          go test -coverprofile="${{ matrix.platform }}_coverage-unit.out" -covermode=atomic -coverpkg=./... ./... --tags=unit
+          go tool cover -func="${{ matrix.platform }}_coverage-unit.out"
 
-      - name: Upload coverage report
+      - name: Run integration tests
+        run: |
+          go test -coverprofile="${{ matrix.platform }}_coverage-integration.out" -covermode=atomic -coverpkg=./... ./... --tags=integration
```
Comment (IGZangelcostales) [inline .github/workflows/test-release.yml:51]: try to use the same format name as KAI:
`coverage-integration-${{ matrix.platform }}.out` y `coverage-unit-${{ matrix.platform }}.out`

Comment (kafkaphoenix) [reply .github/workflows/test-release.yml]: done

File: .github/workflows/test-release.yml [hunk 2]
```diff
@@ -65,10 +76,15 @@ jobs:
...
+      - name: Download unit coverage report
+        uses: actions/download-artifact@v1
+        with:
+          name: coverage-unit-ubuntu-latest
+
+      - name: Download integration coverage report
         uses: actions/download-artifact@v1
         with:
-          name: ubuntu-latest_coverage
+          name: coverage-integration-ubuntu-latest
```
Comment (dfinteligenz) [inline .github/workflows/test-release.yml:87]: You'd also need to change the coverage-integration-${{ matrix.platform }} at this point to download the coverage reports correctly.

Comment (kafkaphoenix) [reply .github/workflows/test-release.yml:87]: we are only doing this for ubuntu-latest

**Review by IGZangelcostales — APPROVED** [review]

**Pull request merged**